### PR TITLE
Apr 2642 calculate working days

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.Database/SFA.DAS.ApplyService.Database.sqlproj [hunk 1]
```diff
@@ -81,12 +81,12 @@
     <Build Include="Tables\Appeal.sql" />
     <Build Include="Tables\ExtractedApplications.sql" />
     <Build Include="Tables\SubmittedApplicationAnswers.sql" />
+    <Build Include="Tables\BankHoliday.sql" />
   </ItemGroup>
   <ItemGroup>
     <PostDeploy Include="Script.PostDeployment1.sql" />
   </ItemGroup>
   <ItemGroup>
-    <None Include="SFA.DAS.ApplyService.Database.publish.xml" />
```
Comment (martin-lush) [inline src/SFA.DAS.ApplyService.Database/SFA.DAS.ApplyService.Database.sqlproj:89]: can you include this one back in

File: src/SFA.DAS.ApplyService.Web.UnitTests/Services/BankHolidayServiceTests.cs [hunk 2]
```diff
@@ -1,27 +0,0 @@
...
-        [TestCase("2021-07-01", 10, "2021-07-15")]
-        [TestCase("2021-08-01", 10, "2021-08-16")]
-        public void GetWorkingDaysAhead_returns_expectedDate(DateTime startDate,int daysAhead,DateTime? expectedDate)
-        {
-            var service = new BankHolidayService();
-            var returnedDate = service.GetWorkingDaysAheadDate(startDate, daysAhead);
-            expectedDate.Should().Be(returnedDate);
-        }
-    }
-}
```
Comment (martin-lush) [inline src/SFA.DAS.ApplyService.Web.UnitTests/Services/BankHolidayServiceTests.cs:27]: I would have kept this but mocked the repository

File: src/SFA.DAS.ApplyService.Database/SFA.DAS.ApplyService.Database.sqlproj (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.Database/SFA.DAS.ApplyService.Database.sqlproj:89]: will do

File: src/SFA.DAS.ApplyService.Web/Startup.cs [hunk 3]
```diff
@@ -259,6 +258,7 @@ private void ConfigureDependencyInjection(IServiceCollection services)
             services.AddTransient<ITaskListOrchestrator, TaskListOrchestrator>();
             services.AddTransient<IOverallOutcomeService, OverallOutcomeService>();
             services.AddTransient<IBankHolidayService, BankHolidayService>();
+            services.AddTransient<IBankHolidayRepository, BankHolidayRepository>();
```
Comment (martin-lush) [inline src/SFA.DAS.ApplyService.Web/Startup.cs:261]: This feels a bit dubious (referencing the data repository in the Web projecy)

File: src/SFA.DAS.ApplyService.Database/SFA.DAS.ApplyService.Database.sqlproj (hunk 1 above)
Comment (martin-lush) [reply src/SFA.DAS.ApplyService.Database/SFA.DAS.ApplyService.Database.sqlproj:89]: it got renamed 🤷

**Review by martin-lush — APPROVED** [review]

**Review by martin-lush — DISMISSED** [review]

**Review by martin-lush — APPROVED** [review]

**Pull request merged**